docs/markdown/tutorial/training.md

```diff
@@ -1,173 +1,437 @@
+
+  - Pifpaf:
+
+    An accurate pose esitmation algorithm that generates high resolution confidence map, featured with high accuracy over low resolution images. Paper is [here](https://openaccess.thecvf.com/content_CVPR_2019/html/Kreiss_PifPaf_Composite_Fields_for_Human_Pose_Estimation_CVPR_2019_paper.html).
+
+- **10 types of common model backbone**
+  - Vgg Backbones:
+    - Vggtiny
+    - Vgg16
+    - Vgg19
+  - Resnet Backbones:
+    - Resnet18
+    - Resnet50
+  - Mobilnet Backbones:
+    - MobilenetV1
+    - MobilenetV2
+    - MobilenetThin (in preset model architecture)
+    - MobilenetSmall (in preset model architecture)   
+    - Dilated Mobilenet (in preset model architecture)
+
+- **2 types of popular dataset**
+  - COCO:
+
+    COCO is a large-scale object detection, segmentation, and captioning dataset. COCO Keypoint Detection Dataset includes 11827 train images, 5000 validation images and 40670 test images with 250000+ poeple with keypoints. reference is [here](https://cocodataset.org/#home).
+    - COCO 2014 (version publicated at 2014)
+    - COCO 2017 (version publicated at 2017)
+
+  - MPII:
+
+    MPII Human Pose Dataset is a state of art benchmark for human pose estimation evaluation, which includes 25000 train images, 3000 validation images, and 7000 test images. reference is [here](https://paperswithcode.com/dataset/mpii).
+
+- **Training Options**
+  - Parallel Training
+  - Backbone Pretraining
+  - Domain Adaptation
+
+- **Extension Interfaces**
+  - Custome dataset
+    - User-add dataset
+
+      User add their self-collected data into preset dataset generation pipeline for train and evaluation.
+
+    - User-defined dataset
+
+      User define their own dataset class to take over the whole dataset generation pipeline.
+
+  - Custome model
+
+      User define thier own model class to take over the model forwarding and loss calculation procedure.
+
+  - Custome pipeline
+
+      User use the provided pre-processors, post-processors and visualizers to assemble their own training or evalution pipeline.
+
 
 ## Integrated pipeline
-Hyperpose extract similiar models into a model class. For now, there are two classes: Openpose classes and Poseproposal classes.
-all model architecture can be devided into one of them.  
+Hyperpose provides integrated train, evaluate and test pipeline with various High-level APIs for user to fast adapt the existing pose estimation algorithms for their specific usage scenario.
+
+The whole procedure can be devided into two parts:
+
+**In the first part**, users use the **set** APIs of the **Config** module to set up the components of the pipeline. User can set up from the general settings such as algorithm type, network architecture and dataset type, to the detailed settings including training batch size, save interval and learning rate etc.
```

Comment:
I think in general people use "``" (code blocks) to mark APIs.

Reply:
make sense